tools/pom.xml

```diff
@@ -100,9 +105,16 @@
                             <artifactSet>
                                 <includes>
                                     <include>org.rogach:scallop_${scala.binary.version}</include>
+                                    <include>com.nvidia:rapids-4-spark-common_${scala.binary.version}</include>
                                 </includes>
                             </artifactSet>
                             <filters>
+                                <filter>
+                                    <artifact>com.nvidia:rapids-4-spark-common_${scala.binary.version}</artifact>
+                                    <includes>
+                                        <include>com/nvidia/spark/rapids/ThreadFactoryBuilder*</include>
```

Comment:
Do we really need to manually filter the classes used here?  That's a maintenance burden.  The shade plugin has a [mechanism to automatically include classes that are referenced](https://maven.apache.org/plugins/maven-shade-plugin/shade-mojo.html#minimizeJar), and if that works we should use it here.  If it doesn't work, it would be nice to comment here in the pom why we're not using that feature.

Reply:
Thanks @jlowe for the suggestion. Using `minimizeJar` works. Verified that only referenced classes are included in the jar.